### PR TITLE
648 missing error msg in login spring sec

### LINKED ISSUE #648: spring security: bad login does not show error message
affects spring and spring boot stack. Manifests when spring security is present.

as shown in #634 

image

view should look like:

image

### PR DESCRIPTION
**Pull request by mrioan**
## How to UAT

- Check that error messages are shown in Spring Security after failed authentication (Spring Boot and bare Spring).

Resolves #648

### CODE REVIEW

**Comment by mraible** [comment]
@mrioan I fixed #661 and pushed the change into this branch.

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java [hunk 1]
```diff
@@ -16,8 +14,11 @@
 
     private static final String INVALID_LOGIN_MESSAGE = "stormpath.web.login.form.errors.invalidLogin";
 
+    private static final String UNSUCCESSFUL_LOGIN_STORMPATH_ERROR = "Login attempt failed because there is no Account in " +
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java]: Should this be an `i18n` key?

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java [hunk 2]
```diff
@@ -43,6 +43,8 @@
     protected RequestFieldValueResolver fieldValueResolver;
     protected List<Field> formFields;
 
+    private final static String SPRING_SECURITY_AUTHENTICATION_FAILED_KEY = "SPRING_SECURITY_AUTHENTICATION_FAILED_MESSAGE";
```
Comment (dogeared) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java]: This exact line is in two different places - `FormController` and `StormpathAuthenticationFailureHandler`. Maybe it should be `public static` in one or the other (or a different place altogether)?

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathAuthenticationFailureHandler.java [hunk 3]
```diff
@@ -39,23 +40,32 @@
  */
 public class StormpathAuthenticationFailureHandler implements AuthenticationFailureHandler {
 
+    private final static String SPRING_SECURITY_AUTHENTICATION_FAILED_KEY = "SPRING_SECURITY_AUTHENTICATION_FAILED_MESSAGE";
```
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathAuthenticationFailureHandler.java]: This exact line is in two different places - `FormController` and `StormpathAuthenticationFailureHandler`. Maybe it should be `public static` in one or the other (or a different place altogether)?

**Comment by dogeared** [comment]
**Fails UAT overall**

**Note**: UAT performed in an Incognito window in Chrome.

Steps to reproduce (reproducible with `Spring Boot Default`, `Spring Security Spring Boot WebMVC`, or `Spring Security WebMVC` examples):

**PASS**: Non-existent Account

1) Browse to localhost:8080/login

    image

2) Enter an account that *does not* exist and hit the `Login` button.

    image

    image

** FAIL**: Existing Account, bad password

1) Browse to localhost:8080/login

    image

2) Enter and account that *does* exist, put in the wrong password, and hit the `Login` button:

    image

    image

**Comment by dogeared** [comment]
Back to you @mraible

**Comment by mraible** [comment]
@dogeared The comments you made seem to be on code that @mrioan wrote. 

@mrioan Do you want to fix these issues or would you like me to try?

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java (hunk 1 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java]: This is the error that will be received from the Backend. This text is used in the form of `e.getMessage().contains(UNSUCCESSFUL_LOGIN_STORMPATH_ERROR)` in order to know that an authentication error happened in the backend and thus create our own message.

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java (hunk 2 above)
Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java]: You are right. Done

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java (hunk 1 above)
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/SpringSecurityLoginErrorModelFactory.java]: I just changed this text to `Login attempt failed`. That catches both "account does not exist" and "account does exist but password is wrong" (the latter was your finding in UAT). This change fixes that. Thanks!

**Comment by mrioan** [comment]
@dogeared thanks for finding the `Login attempt failed because the specified password is incorrect` error. I introduced a fixed for it.

**Comment by dogeared** [comment]
I confirmed that the issue @mraible outlined above is fixed. Please merge @mrioan.

**Pull request merged**